far/help.cpp

```diff
@@ -1453,7 +1453,8 @@ bool Help::ProcessKey(const Manager::Key& Key)
 			// не поганим SelTopic, если и так в FoundContents
 			if (StackData.strHelpTopic != FoundContents)
 			{
-				if (GetSearchReplaceString({}, m_SearchDlgParams, L"HelpSearch"sv, {}, CP_DEFAULT, {}, &HelpSearchId) == SearchReplaceDlgResult::Cancel)
+				if (GetSearchReplaceString({ .AllowEmptyPattern = true }, m_SearchDlgParams, L"HelpSearch"sv, {}, CP_DEFAULT, {}, &HelpSearchId)
```

Comment:
Why do we allow this in Help?

Reply:
I'm guessing because an empty search displays all topics from the _Help_, not just the ones listed in _Contents_. I have used this feature in the past.

Reply:
Done!

Reply:
:bulb: If anyone is interested in seeing the old format, the one that showed up when searching with an empty string, just search for `#`. You'll get the same list as before, with all the topics listed in the same order as the `.hlf` _Help_ file.